mongo/integration/unified/cursor_entity.go

```diff
@@ -0,0 +1,22 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package unified
+
+import (
+	"context"
+)
+
+// The cursor interface defines the methods that must be implemented by iterable types that can be stored in an
+// EntityMap. This type exists so that we can store multiple concrete types returned by functions from the mongo package
+// under a single interface (e.g. mongo.Cursor and mongo.ChangeStream).
```

Comment:
Is there anything else that might implement `cursor` besides `Cursor` and `ChangeStream`?

Reply:
Nothing right now, but any cursor-like objects we may add to the `mongo` package in the future should follow this basic interface anyway for consistency. If we get to a point where we need to add such an object with a drastically different interface, we can revisit and figure out what to do here (e.g. we might need to add another entity type or create a wrapper type in the test runner code that does implement this interface).

Reply:
Sounds good.